### ISSUE TITLE
Dependency on eudev

### ISSUE THREAD

**Issue by Riteo** [kiss-community/community#731]
Hi! I found out that a few packages depend on `eudev`:
 - `libgphoto2`
 - `chromium` (not really in the `depends` file, but `post-install` advises installing it or `udev-zero` in order to pass some checks)
 - `tlp`
 - `usbutils`

 I tried removing the `eudev` dependency in `usbutils` and it works fine. Since it's not immediately clear on why those packages depend on it and installing those (pretty popular) packages would require swapping a relatively important system component, would it be possible to document (or at least explain as a comment here) why is `eudev` needed as a dependency of these packages, maybe even in the proposed per-package `README` file?

**Comment by dilyn-corner** [comment]
This would be up to the maintainers to answer:
`libgphoto2` -- @vouivre
`tlp` -- @depsterr
`usbutils` -- @trbednarzyk

**Comment by depsterr** [comment]
TLP is no longer maintained by me, I recall leaving it up for grabs, either I'm misstaken or no ine has picked it up

**Comment by Vouivre** [comment]
I will have a look at the README when I will have time. In between, `libgtphoto2` creates some `udev` rules. With
those rules, the user can have access to the camera, otherwise only root can access the camera. It only works if
`eudev` is installed. Perhaps a more lightweight solution is possible (`mdevd` for example). I'm interested by any suggestion.

**Comment by Riteo** [comment]
@Vouivre I've found a way to achieve something similar, although I haven't still tried it for your package. `android-tools` accesses its android devices directly through the USB and due to this, can't work as is without running its server as root. For this exact reason `android-udev` exists to provide rules for providing access to normal users by checking for a specific vendor and product id and `chmod`ing the device's USB port device file. Apparently this is easily doable by detecting all usb events using the command syntax (see [mdev's documentation](https://git.busybox.net/busybox/plain/docs/mdev.txt) and, for a more real example, [this mdev-like-a-boss's config line](https://github.com/slashbeast/mdev-like-a-boss/blob/11c3ff31a2da299ca0811158e41a9652f5d614be/mdev.conf#L108)) that, through its enviroment variables updated by mdev, checks for the right vendor and product id and `chmod`s it.

Speaking in real terms:

`mdev.conf` (not the whole file, just the addition):
```
# Run /etc/mdev.d/android-mdev (could be anywere) on any usb insertion event.
`SUBSYSTEM=usb;.* root:root 660 ! @/etc/mdev.d/android-mdev`
```

`/etc/mdev.d/android-mdev`:
```
#! /bin/sh
# Just an example, filters a single (null) device.
# The string that gets set on PRODUCT has the following format:
# vendorId/productId/revision
#
# DEVNAME is a path relative to /dev to the device file.
# These enviroment variables get set by mdev from the USB uevents, look at them
# for more information.

if [ "$PRODUCT" = "0000/0000/000" ]
then
	chmod g+rw "/dev/$DEVNAME" 
	chgrp android "/dev/$DEVNAME"
fi
```

This is just a proof of concept, but this mechanism has a lot of potential. IMO the script should read by itself a separate list of allowed devices, in order to avoid dependencies on specific device managers, but this is up for discussion. Maybe this could even be included in the core busybox package?

**Comment by dilyn-corner** [comment]
Correct, a large number of udev rules are simply to set proper permissions on device paths -- if these packages only require `eudev` for this functionality, this can be accomplished (perhaps nontrivially) in many different ways. For instance, the kernel has a builtin way of doing this (`UEVENT_HELPER`). `libudev-zero` includes `helper.c` (though I'm not certain how its functionality has changed since the 1.0 release). 

We already suggest taking @riteo's suggestion -- see how Dylan handles `libudev-zero` and the `mdev.conf` here: https://github.com/kisslinux/repo/tree/master/extra/libudev-zero (the `mdev.conf` suggestion already exists in the `busybox` package: https://github.com/kisslinux/repo/blob/master/core/busybox/files/mdev.conf#L72-L74). So we have two options in these cases:
1) Pester Dylan with a PR to add these changes to the `mdev.conf` included with `busybox`
2) Tell users what to do in something like a `post-install` file OR do it for them

The first option only helps people who use a 'default' KISS system. Option 2 would solve this problem on a per-package basis, which is probably preferred (the main repo should only really include what's necessary to "get off the ground" with KISS, not "extras" like Android devices when you have to install a package for that on top of what `busybox` is providing). 

As for whether it's done automatically for users (like https://github.com/kiss-community/community/blob/main/community/oksh/post-install) as opposed to manually by users (like https://github.com/kiss-community/community/blob/main/community/openntpd/post-install) is up to the maintainer IMHO.

**Comment by Riteo** [comment]
@dilyn-corner the issue with auto install or even just manual configuration is that it's too unstructured by itself. What I proposed to include in `repo` was really just a script/folder and its respective hooks already setup by default for a more modular and less error prone device hook system. Kinda like dylan's boot manager agnostic boot system. I am aware of solutions like uevent or even the kernel's own `hotplug` system. The idea behind this is that, being uevent arguments consistent, as long ad a device manager can run scripts it should be a agnostic as it can be, even by just running `uevent` through it.

**Comment by dilyn-corner** [comment]
Let me rephrase what I was trying to say:

I don't know what your suggestion is accomplishing that isn't already solved, and the original aspect of this issue ("why do these things depend on `eudev`") is resolved by my suggestion to fill the gap on the already existing solution
> Tell users what to do in something like a post-install file OR do it for them

**Comment by Vouivre** [comment]
Sorry for not responding, I need a better understanding in this issue, but at the moment I can't. If something
must be done on my package, you can make the change without problem. I expect I won't be able to have a better
look before a few weeks.

**Comment by Vouivre** [comment]
I'm back in this thread, sorry for the delay. @Riteo can you confirm me there is nothing to do on the package at the moment until you contact Dylan ? 
To be honest, I still don't have an overview of the possibilities to solve the issue. I only have a better 
understanding of `mdev`.

**Comment by Riteo** [comment]
@Vouivre yes, exactly. Actually I don't think that a deep understanding of eudev will be actually needed, so don't worry. :)

**Comment by Vouivre** [comment]
@Riteo Thank you!

**Comment by dilyn-corner** [comment]
Assuming `libgphoto2` is trapped forever with `eudev`, considering this closed.

**Comment by Riteo** [comment]
Yeah this issue can't really be fixed without some new radical solution made by someone and miraculously accepted by everyone or by Dylan and his public return. I agree on your decision.

**Comment by git-bruh** [comment]
Continued in https://github.com/kiss-community/repo/issues/89
